### ISSUE TITLE
MonitoringFactory does not trigger callbacks for PutObject and GetObject

### ISSUE THREAD

**Issue by VasylHryhorzhevskyi** [aws/aws-sdk-cpp#3135]
### Describe the bug

Used C++ SDK version 1.11.407
During the application working there was calls fro requests:
HeadBucket
CreateSession
CreateBucket
ListObjectsV2
DeleteObjects

But for previous version of SDK there was triggered also:
PutObject
GetObject

Also we have moved from cUrl and WinHttp  to S3Crt and CrtHttp.

Do you need something else?

### Expected Behavior

MonitoringFactory triggers:
PutObject
GetObject

### Current Behavior

MonitoringFactory does not trigger:
PutObject
GetObject

### Reproduction Steps

1) use MonitoringFactory for the application
2) do PutObject or GetObject
3) check requestName in the MonitoringFactory::OnRequestFailed or MonitoringFactory::OnRequestSucceeded functions calls.

Currently step 3 does not happens.

### Additional Information/Context

Application build for Mac Win and Al2 operations systems.

### AWS CPP SDK version used

1.11.407

### Compiler and Version used

clang-1500.3.9.4, 

### Operating System and version

Windows_Server-2022-English-Full-Base-2024.06.13, amzn-ec2-macos-14.6.1-20240814-230108, Amazon Linux 2 AMI (HVM) - Kernel 5.10

**Comment by sbera87** [comment]
On checking the code, PutObject and GetObject for crt client doesn't invoke the monitoring callback methods. We are assessing the best way to address this

**Comment by sbera87** [comment]
> On checking the code, PutObject and GetObject for crt client doesn't invoke the monitoring callback methods. While we explore how to add support for this, please also consider using telemetry provider already available in client configuration.
> 
> https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/include/aws/core/client/ClientConfiguration.h#L417-L421

Monitor support for s3 crt client get/put object is underway.

**Comment by VasylHryhorzhevskyi** [comment]
We use AWS monitoring to get the following information:
1. Status of individual request (re)try (success/failure)
2. Thread ID (or similar information about context to distinguish interleaved log messages from requests that complete in parallel)
3. Request name (PutObject, GetObject, HeadBucket, etc)
4. Number of attempts after which request succeeded or failed
5. HTTP response code
6. Remote IP
7. Whether request is retry-able (see AWSError<T>::ShouldRetry())
8. AWSError<T>::GetErrorType()
9. AWSError<T>::GetExceptionName()
10. AWSError<T>::GetMessage()
11. AWS request ID (see AWSError<T>::GetRequestId())
12. Duration of individual request (re)try and total duration of request (all retries added up)
13. HTTP client metrics (connect latency, ssl latency, etc)
14. Request headers
15. Response headers
16. Response XML payload
17. Request URI (for PutObject and GetObject it includes object name)

@sbera87 could you provide a sample code for telemetry provider that demonstrates how to obtain this information.

**Comment by sbera87** [comment]
So, telemetry provides tracer (for logging data points) and a meter (for statistics on data points).
You need to use this with the the utils: https://github.com/aws/aws-sdk-cpp/blob/c5eb78b5e3befd1d3345ff94d7c9458bcea59fc6/src/aws-cpp-sdk-core/include/smithy/tracing/TracingUtils.h#L21

Span: src/aws-cpp-sdk-core/include/smithy/tracing/TraceSpan.h

Meter: src/aws-cpp-sdk-core/include/smithy/tracing/Meter.h

Example usage in sdk:
Constructor for telemetry provider: https://github.com/aws/aws-sdk-cpp/blob/c5eb78b5e3befd1d3345ff94d7c9458bcea59fc6/src/aws-cpp-sdk-core/source/client/AWSClient.cpp#L124 

Usage of meter:
https://github.com/aws/aws-sdk-cpp/blob/c5eb78b5e3befd1d3345ff94d7c9458bcea59fc6/generated/src/aws-cpp-sdk-dynamodb/source/DynamoDBClient.cpp#L237

How sdk measures http metrics: https://github.com/aws/aws-sdk-cpp/blob/c5eb78b5e3befd1d3345ff94d7c9458bcea59fc6/src/aws-cpp-sdk-core/source/client/AWSClient.cpp#L308

It does mean that with telemetry provider you need to implement those hooks that are called in the sdk

**Comment by sbera87** [comment]
Monitoring has been added for the concerned S3 APIs. Will close this for now . If any issue faced, please reopen this or create a new issue.
